src/nix/command.hh

```diff
@@ -70,6 +70,8 @@ struct EvalCommand : virtual StoreCommand, MixEvalArgs
 {
     ref<EvalState> getEvalState();
 
+    ~EvalCommand();
```

Comment:
The problem with printing the warning from the destructor is that you only get the warning *after* building, which could be a long time. You don't want to have Nix build stuff for an hour to then find out that your flag was ignored. Maybe the check can be done after `resolveFlake()` (e.g. in `InstallableFlake`)?